File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for bracket in s :
        if len(stack) == 0 or (stack[-1] + bracket != "()") :
            stack.append(bracket)
            continue
        stack.pop()
    
    if len(stack) == 0:
        return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;
class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ')') {
                if (stack.empty()) return false;
                else {
                    stack.pop();
                    continue;
                }
            }
            
            stack.push(s.charAt(i));
        }
        
        return stack.empty();
    }
}